### DISCUSSION TITLE
How do I try mpl2 on a design?

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD-flow-scripts#953] [General]
I'm curious about mpl2....

What do I have to do to try mpl2 on e.g. designs/asap7/mock-array-big?

mock-array-big uses manual macro placement and places the macros into a grid. I'm curious if mpl2 can figure out that a grid is the right layout for mock-array-big as well as trying out mpl2 for other designs as mpl (the current default) doesn't come up with macro placements that are going to work in my cases.

From searching, [guessed](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/954), but mpl2 appears to be crashing.

**Comment by vijayank88** [comment]
@oharboe 
in your `config.mk` add following lines and execute, it will use `mpl2` feature for macro placement.
```
export SYNTH_HIERARCHICAL = 1
export RTLMP_FLOW = True
```
Also pin placement restricted on either side with following constraints:
```
export HAS_IO_CONSTRAINTS = 1
export PLACE_PINS_ARGS = -exclude left:* -exclude right:* -exclude top:*
```

**Reply by oharboe** [reply]
Thanks! Trying [here](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/954)

Though... I don't understand what you mean by IO pin constraints. mock-array-big already has io pin constraints that make sense to lay out things in an array.

**Reply by oharboe** [reply]
@vijayank88 Any idea why mpl2 crashes in #954 ?

**Reply by vijayank88** [reply]
@ravi-varadarajan Can you please check this?

**Reply by ravi-varadarajan** [reply]
> @ravi-varadarajan Can you please check this?

Is this test case already in ORFS?

**Reply by vijayank88** [reply]
Yes. Asap7/mock-big-array
